[Darryl Chan] iP

darryl-chan opened the pull request:
# LinusBot

> "Better responses than ChatGPT" - OpenAI CEO

Have you ever wanted a bot that is
-  Text-based
-  Keeps track of your tasks
-  ~~Easy~~ Very easy to use

Then you came to the right place 😉 

You just have to:
1. Download it [here](https://github.com/darryl-chan/ip)
2. run it
3. Its that easy

And might I remind you that its **FREE** to use

Current features:
- [x] Manage tasks
- [x] Cool graphics

And if you love java you can download this app and improve your skills. Here's the `Launcher` class:

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

darryl-chan committed:
Testing git

darryl-chan committed:
Level 0

darryl-chan committed:
Added greet and goodbye

darryl-chan committed:
Added background program for chatbot

darryl-chan committed:
Merge branch 'master' of https://github.com/darryl-chan/ip

darryl-chan committed:
Added echoing

darryl-chan committed:
Added List and its functions

darryl-chan committed:
Added tasks and task markers

darryl-chan committed:
Added Todo, event, deadline

darryl-chan committed:
Fixed display bugs for tasks

darryl-chan committed:
Added I/O testing

darryl-chan committed:
Added some exception handling

darryl-chan committed:
Added a delete function

darryl-chan committed:
Updated Linus logo

darryl-chan committed:
Added switch cases and refined error messeges

darryl-chan committed:
Updated I/O

darryl-chan committed:
combined mark and delete functions and cleaned code

darryl-chan committed:
Added switch cases for event creation

darryl-chan committed:
Merge pull request #1 from darryl-chan/better-user-input-function

Added switch cases and better exception handling

darryl-chan committed:
Added saving file function

darryl-chan committed:
Added string split

darryl-chan committed:
Added load file function and dealing with corrupted file

darryl-chan committed:
Merge branch 'branch-Level-7'

darryl-chan committed:
Added date time functions

darryl-chan committed:
added save format

darryl-chan committed:
Created save file format in tasks

darryl-chan committed:
Edited saving file format

darryl-chan committed:
Moved duplicated codes in deadline and event to datetask

darryl-chan committed:
Changed the format of how the dates printed

darryl-chan committed:
edited linus.txt

darryl-chan committed:
Merge branch 'branch-level-8'

darryl-chan committed:
Changed PrintList to Ui

darryl-chan committed:
Moved some prints to UI

darryl-chan committed:
cleaned code

darryl-chan committed:
Moved programs to duke and tasks

darryl-chan committed:
Added to packages

darryl-chan committed:
Merge branch 'add-gradle-support'

darryl-chan committed:
Merged gradle support

darryl-chan committed:
Added storage class

darryl-chan committed:
Moved storage functions in program to storage

darryl-chan committed:
Added commands for {BYE, LIST, MARK, UNMARK, DELETE}

darryl-chan committed:
Added Command classes

darryl-chan committed:
Moved tasklist methods to storage

darryl-chan committed:
Created package for Commands

darryl-chan committed:
Added more exceptions and moved them to packages

darryl-chan committed:
Merge branch 'make-oop'

darryl-chan committed:
Moved TaskList to tasks

darryl-chan committed:
Renamed command folder to commands

darryl-chan committed:
Added unit tests for TaskList, Task and DateTask and add toString method for TaskList

darryl-chan committed:
Added CommandFind

darryl-chan committed:
Added javaDoc to commands

darryl-chan committed:
Added JavaDoc for exceptions

darryl-chan committed:
Added JavaDoc for Tasks

darryl-chan committed:
Added JavaDocs to Duke, Parser, Storage, Ui

darryl-chan committed:
Enforced checkstyle

darryl-chan committed:
Merge branch 'branch-A-CodingStandard'

darryl-chan committed:
Merge branch 'branch-A-JavaDoc'

darryl-chan committed:
Merged some conflicts

darryl-chan committed:
Merged conflicts

darryl-chan committed:
Updated gradle build to accomodate java.FX

darryl-chan committed:
Added base for GUI

darryl-chan committed:
Fixed GUI bugs

darryl-chan committed:
Added basic GUI response from Duke

Kappaccinoh commented on docs/README.md:
```diff
@@ -1,29 +1,30 @@
-# User Guide
+# Duke User Guide
```
I see that you have actually renamed your ChatBot to Linus! It's a nice name! Might it be better to change your main class to Linus to avoid confusion with Duke?

Kappaccinoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,205 @@
+package duke;
```
I see that you have actually renamed your ChatBot to Linus! It's a nice name! Might it be better to change your main class to Linus to avoid confusion with Duke?

Kappaccinoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,205 @@
+package duke;
+
+// import javafx.scene.control.Label;
```
I understand that you have moved these comments out to another directory, I think you can delete these comments to keep the file cleaner and shorter

Kappaccinoh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,113 @@
+ */
+public class Parser {
+
+    enum CommandType {LIST, BYE, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT, FIND};
```
Nice use of Enums!

Kappaccinoh commented on src/main/java/duke/commands/CommandBye.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
cool - I didnt know you could do this

Kappaccinoh reviewed (commented):
Great job on the iP thus far, it seems like you have covered the tasks needed. Overall no complaints about coding standards. You have added javadocs, named your variables correctly, organised your files in a manner that is noticeably consistent (and is hence readable in quite a quick fashion because I know where to look), and the names of the classes are also well named and as a result self explanatory!

darryl-chan committed:
Adjusted main window to be bigger and changed label to text area

darryl-chan committed:
Readded the option to load and save

darryl-chan committed:
Merge branch 'Level-10'

darryl-chan committed:
Added assertions, to Parser.java and Storage.java to ensure there is a user input and ensure that directory exists

darryl-chan committed:
Improved code quality by refactoring many ui phrases in Commands and Tasks

darryl-chan committed:
Merge pull request #2 from darryl-chan/branch-A-Assertions

Add assertions to Parser.java and Storage.java

darryl-chan committed:
Merge pull request #3 from darryl-chan/branch-A-CodeQuality

Improve code quality by refactoring methods many ui phrases in Commands and …

darryl-chan committed:
Add workflow for gradle build and check

darryl-chan committed:
Merge branch 'branch-A-CI'

darryl-chan committed:
Add sort for tasks by description, tasktypes, mark and dates

darryl-chan committed:
Merge branch 'branch-C-Sort'

darryl-chan committed:
Remove enums as the default case is not reached

Parser: default case can never be reached because there is try catch
block that catches unknown inputs and making the code non-functional

Now the enums commands have been removed and it relies on string inputs
only

darryl-chan committed:
Remove enums as the default case is not reached

darryl-chan committed:
Merge branch 'master' of https://github.com/darryl-chan/ip

darryl-chan committed:
Add new exceptions so that it is thrown instead of DukeCeption

Previously, unknown commands had no exception of their own and such DukeCeption was thrown instead which was not specific

I have created a class to deal with unknown commands so that the method will throw UnknownCommandException instead

darryl-chan committed:
Add new images for Linus and User, cleaned the code abit and added new dialogue box at the start of the app

darryl-chan committed:
Merge branch 'A-Personality'

darryl-chan committed:
Rename Duke as Linus for all files

darryl-chan committed:
Remove unused imports in J-unit tests

darryl-chan committed:
Revert text area to label as it wraps text better

darryl-chan committed:
Remove redundant files

darryl-chan committed:
Update README.md

darryl-chan committed:
Update README.md

darryl-chan committed:
Merge pull request #5 from darryl-chan/darryl-chan-patch-1

Darryl chan patch 1

darryl-chan committed:
Update code to follow coding standard

darryl-chan committed:
Update gradle build

darryl-chan committed:
Update readme file

darryl-chan committed:
Remove dead codes

darryl-chan committed:
Rename ui.png to Ui.png

darryl-chan committed:
Fix ui bugs, create smaller window for smaller screens

darryl-chan committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

darryl-chan committed:
Fix printing out loading file when exit

darryl-chan committed:
Update loading message
